code/framework/api/src/main/java/io/cattle/platform/api/auth/impl/DefaultPolicy.java

```diff
@@ -67,4 +71,23 @@ public String getUserName() {
         return name;
     }
 
+    @Override
+    public <T> void grantObjectAccess(T obj) {
+        ApiRequest apiRequest = ApiContext.getContext().getApiRequest();
+        @SuppressWarnings("unchecked")
+        Set<Object> whitelist = (Set<Object>) (apiRequest.getAttribute("whitelist"));
+        if(whitelist == null) {
+            whitelist = new HashSet<>();
+        }
+        whitelist.add(obj);
+        apiRequest.setAttribute("whitelist", whitelist);
+    }
+    
+    protected <T> Boolean hasGrantedAccess(T obj) {
```

Comment:
Can you switch this to boolean, you don't need to to force callers to check for null.